lms/djangoapps/ccx/views.py

```diff
@@ -85,20 +86,23 @@ def wrapper(request, course_id):
             ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
             course_key = ccx.course_id
 
-        role = CourseCcxCoachRole(course_key)
-        if not role.has_user(request.user):
-            return HttpResponseForbidden(
-                _('You must be a CCX Coach to access this view.'))
-
         course = get_course_by_id(course_key, depth=None)
-
-        # if there is a ccx, we must validate that it is the ccx for this coach
-        if ccx is not None:
-            coach_ccx = get_ccx_for_coach(course, request.user)
-            if coach_ccx is None or coach_ccx.id != ccx.id:
-                return HttpResponseForbidden(
-                    _('You must be the coach for this ccx to access this view')
-                )
+        is_staff = has_access(request.user, 'staff', course)
+        is_instructor = has_access(request.user, 'instructor', course)
+
+        if is_staff or is_instructor:
+            # if user is staff or instructor then he can view ccx coach dashboard.
+            return view(request, course, ccx)
+        else:
+            role = CourseCcxCoachRole(course_key)
+            if not role.has_user(request.user):
+                return HttpResponseForbidden(_('You must be a CCX Coach to access this view.'))
+
+            # if there is a ccx, we must validate that it is the ccx for this coach
+            if ccx is not None:
+                coach_ccx = get_ccx_for_coach(course, request.user)
+                if coach_ccx is None or coach_ccx.id != ccx.id:
+                    return HttpResponseForbidden(_('You must be the coach for this ccx to access this view'))
```

Comment:
Is there a reason that a HttpResponse object is returned vs raising a PermissionDenied exception? 

It would prefer the PermissionDenied exception was raised instead of returning a HttpResponse object. this will ensure that the wrapper only returns a single type.